internal/testing/factories/factories.go

```diff
@@ -0,0 +1,50 @@
+package factories
+
+import (
+	"context"
+	"crypto/rand"
```

Comment:
This is an overkill, you can use just rand package which gives pseudo random numbers instead. Tests could deplete good entropy just for nothing and a CI instance might get short of good entropy for no reason. The key generator works with any reader.

Reply:
could you pls point me to that package? I need a reader, which math/rand does not have and I think that crypto does not use enthropy by default, or does it?

Reply:
I thought this would be sufficient: https://pkg.go.dev/math/rand

I *think* it has the same interface, if not, keep using cryptographic (e.g. much stronger) generator. It's a nitpick.

Reply:
yeah I've tried that one, but it does not have the function [`Read`](https://pkg.go.dev/crypto/rand@go1.19#Read) I actually use unfortunatelly... the interfaces are not the same (I'd guess as purpose is not the same, they just don't try).
If that's ok with you, I'd keep crypto